jupyterhub/utils.py

```diff
@@ -371,6 +372,11 @@ def compare_token(compare, token):
     return False
 
 
+def url_escape_path(value):
+    """Escape a value to be used in URLs, cookies, etc."""
+    return quote(value, safe='@~')
```

Comment:
This is taken from https://github.com/jupyterhub/jupyterhub/blob/b18a05c2c8a8ff86f9255bc2abaa356c418d3dd3/jupyterhub/user.py#L495-L497